controllers/openshift_controller.go

```diff
@@ -507,7 +508,16 @@ func (r *KataConfigOpenShiftReconciler) getExtensionName() string {
 		return "sandboxed-containers" // RHCOS
 	}
 
-	// As RHCOS is rather special variant, use "kata-containers" by default, which also applies to FCOS
+	if strings.HasPrefix(clusterVersion.Status.Desired.Image, "quay.io/okd/scos-release") {
+		return "kata-containers" // SCOS
+	}
+
+	cmdline, err := ioutil.ReadFile("/proc/cmdline")
+	if err == nil && strings.Contains(string(cmdline), "ostree/rhcos") {
+		return "sandboxed-containers" // RHCOS
+	}
```

Comment:
While I appreciate the reasoning I'd actually consider dropping the `clusterVersion.Status.Desired.Image` checks.  If they don't work then the OKD team endorsement seems of little value.

We can keep them if the kernel cmdline based check is not reliable either and the other checks cover the cases where it fails.

Reply:
Fair enough but let's make a solemn vow here that we'll change the RHCOS extension name ASAP, ideally pretty much immediately, shall we?  Otherwise I'd have to pity the poor newcomer soul trying to make sense of this cascade of checks. ;-)

Reply:
> Fair enough but let's make a solemn vow here that we'll change the RHCOS extension name ASAP, ideally pretty much immediately, shall we? Otherwise I'd have to pity the poor newcomer soul trying to make sense of this cascade of checks. ;-)

I do solemnly swear that I will faithfully create a PR in the MCO repo, and will to the best of my ability, push, argue and merge a new extension name for kata.